components/popover_list_members/popover_list_members.jsx

```diff
@@ -97,15 +96,15 @@ export default class PopoverListMembers extends React.Component {
         const isChannelAdmin = ChannelStore.isChannelAdminForCurrentChannel();
         const membersIcon = Constants.MEMBERS_ICON_SVG;
 
-        if (members && teamMembers) {
-            members.sort((a, b) => {
-                const aName = Utils.displayUsernameForUser(a);
-                const bName = Utils.displayUsernameForUser(b);
-
-                return aName.localeCompare(bName);
+        if (this.props.members && teamMembers) {
+            const sortedMembers = this.props.members.map((member) => {
+                const status = UserStore.getStatus(member.id);
+                return {...member, status};
+            }).sort((a, b) => {
+                return UserStatusesWeight[a.status] - UserStatusesWeight[b.status] || Utils.sortUsersByDisplayName(a, b);
             });
```

Comment:
Can this sorting be generalized somewhere so that we aren't duplicating the code in the two components?